keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/InverseGammaVertexTest.java

```diff
@@ -0,0 +1,174 @@
+package io.improbable.keanu.vertices.dbl.probabilistic;
+
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import org.apache.commons.math3.stat.descriptive.SummaryStatistics;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import static io.improbable.keanu.vertices.dbl.probabilistic.ProbabilisticDoubleContract.moveAlongDistributionAndTestGradientOnARangeOfHyperParameterValues;
+import static org.junit.Assert.assertEquals;
+
+public class InverseGammaVertexTest {
+
+    private final Logger log = LoggerFactory.getLogger(InverseGammaVertexTest.class);
+
+    private Random random;
+
+    private static final double DELTA = 0.001;
+
+    @Before
+    public void setup() {
+        random = new Random(3);
+    }
+
+    @Test
+    public void samplingProducesRealisticMeanAndStandardDeviation() {
+        int N = 100000;
```

Comment:
Can we factor this into ProbabilisticDoubleContract? We seem to be doing this for each vertex.

Reply:
Good idea, will do